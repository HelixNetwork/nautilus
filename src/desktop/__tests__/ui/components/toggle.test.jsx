import React from 'react';
import { shallow } from 'enzyme';

import Toggle from 'ui/components/toggle';

const props = {
    checked: false,
    on: 'On',
    off: 'Off',  
    onChange: jest.fn(),
};

describe('Test for Toggle component', () => {
    test('Render the Toggle component', () => {
        const wrapper = shallow(<Toggle {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Checks Toggle Off state', () => {
        const wrapper = shallow(<Toggle {...props} />);

        expect(wrapper.find('div div').hasClass('on')).toBeFalsy();
    });

    test('Checks Toggle On state', () => {
        const mockProps = Object.assign({}, props, { checked: true });
        const wrapper = shallow(<Toggle {...mockProps} />);

        expect(wrapper.find('div div').hasClass('on')).toBeTruthy();
    });

    test('Checks the Toggle event', () => {
        const wrapper = shallow(<Toggle {...props} />);

        wrapper.simulate('click');
        expect(props.onChange).toHaveBeenCalledTimes(1);
    });
});
