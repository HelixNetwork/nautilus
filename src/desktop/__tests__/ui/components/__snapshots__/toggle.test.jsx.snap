// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Toggle component Render the Toggle component 1`] = `
<div
  className="toggle"
  onClick={[Function]}
>
  <div
    className="active"
  >
    <span>
      Off
    </span>
    <span>
      On
    </span>
  </div>
</div>
`;
